cardano-api/src/Cardano/Api/MetaData.hs

```diff
@@ -78,44 +78,35 @@ validateTxMetadata
   :: TxMetadata
   -> Either TxMetadataValidationError TxMetadata
 validateTxMetadata txMd@(TxMetadata (MetaData mdMap)) =
-    Map.foldl' foldFn (Right txMd) mdMap
+    txMd <$ traverse_ validate (Map.elems mdMap)
   where
-    foldFn
-      :: Either TxMetadataValidationError TxMetadata
-      -> MetaDatum
-      -> Either TxMetadataValidationError TxMetadata
-    foldFn lastResult metaDatum =
-      case lastResult of
-        Left err -> Left err
-        Right _ -> validate metaDatum
-
-    validate :: MetaDatum -> Either TxMetadataValidationError TxMetadata
+    validate :: MetaDatum -> Either TxMetadataValidationError ()
```

Comment:
`Right txMd` was returned for every occurrence of `Right`, so no information is conveyed by the payload of `Right`.

I changed the type signature to return `Either TxMetadataValidationError ()` to reflect this.